src/lib/AJAX.js

```diff
@@ -9,8 +9,22 @@ import * as CSRFManager from 'lib/CSRFManager';
 import encodeFormData   from 'lib/encodeFormData';
 import { Promise }      from 'parse';
 
+let basePath = '';
+export function setBasePath(newBasePath) {
+  basePath = newBasePath || '';
+  if (basePath.endsWith('/')) {
+    basePath = basePath.slice(0, basePath.length-1);
+  }
+}
+
 // abortable flag used to pass xhr reference so user can abort accordingly
 export function request(method, url, body, abortable = false, withCredentials = true, useRequestedWith = true) {
+  if (!url.startsWith('http://')
```

Comment:
Yeah this is fine. The only things that aren't going to go through the Parse sdk are the config loading, and potentially this app creation stuff that is being discussed.